src/features/class-graphql.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * GraphQL class file
+ *
+ * @package wp-curate
+ */
+
+namespace Alley\WP\WP_Curate\Features;
+
+use Alley\WP\Types\Feature;
+use WPGraphQL\AppContext;
+use GraphQL\Type\Definition\ResolveInfo;
+use WPGraphQL\Data\Connection\PostObjectConnectionResolver;
+
+/**
+ * Add support for GraphQL, if WPGraphQL plugin exists.
+ */
+final class GraphQL implements Feature {
+	/**
+	 * Post types allowed in GraphQL.
+	 */
+	private $allowed_post_types;
+
+	/**
+	 * Get the GraphQL Type by Post Type.
+	 *
+	 * @param  string $post_type_string Name of the post type. Ex: 'post'.
+	 *
+	 * @return string GraphQL Type name. Ex: 'Post'.
+	 */
+	private function get_graphql_type_by_post_type( $post_type_string ): string {
+		$post_type_object = get_post_type_object( $post_type_string );
+
+		if ( empty( $post_type_object ) ) {
+			return '';
+		}
+
+		// Only return a GraphQL type for allowed post types in WP Curate.
+		if ( ! in_array( $post_type_string, $this->allowed_post_types, true ) ) {
+			return '';
+		}
+
+
+		return ucfirst( $post_type_object->graphql_single_name );
+	}
+
+	/**
+	 * Get GraphQL Types from allowed post types.
+	 *
+	 * @return array<string>
+	 */
+	private function get_types_from_allowed_post_types(): array {
+		$interface_to_types = [];
+
+		foreach ( $this->allowed_post_types as $post_type ) {
+			$interface_to_types[] = $this->get_graphql_type_by_post_type( $post_type );
+		}
+
+		return $interface_to_types;
+	}
+
+	/**
+	 * Set up.
+	 */
+	public function __construct() {
+		$this->allowed_post_types = apply_filters( 'wp_curate_allowed_post_types', [ 'post', 'opinion' ] );
+	}
+
+	/**
+	 * Boot the feature.
+	 */
+	public function boot(): void {
+		// Assumes that WPGraphQL has been installed as a composer dependency of a parent project.
+		if ( class_exists( 'WPGraphQL' ) ) {
+			add_action( 'graphql_register_types', [ $this, 'graphql_register_types' ] );
+		}
```

Comment:
The way the hook system works means that we don't actually need to see if the WPGraphQL class exists at all, we can register a callback for a graphql hook, which won't fire if the plugin isn't installed and activated:
```suggestion
		add_action( 'graphql_register_types', [ $this, 'graphql_register_types' ] );
```

Reply:
Ah yes, makes sense.